### PR TITLE
Implement From<String> for Atom

### PR DESCRIPTION
**Pull request by nox**
[Review on Reviewable](https://reviewable.io/reviews/servo/string-cache/143)

### CODE REVIEW

**Comment by SimonSapin** [comment]
`From<&'a str> for Atom` now has an unnecessary `str::to_owned` call for dynamic atoms that are already in the hash table. Could `StringCache::add` take a `Cow<str>` instead of `String`?

**Comment by nox** [comment]
Yeah that was an error on my part.

File: src/atom/mod.rs [hunk 1]
```diff
@@ -171,6 +178,34 @@ impl<'a> From<&'a str> for Atom {
...
+}
+
+impl From<String> for Atom {
+    #[inline]
+    fn from(string_to_add: String) -> Atom {
+        Atom::from(Cow::Owned(string_to_add))
+    }
+}
+
+impl From<Atom> for String {
```
Comment (SimonSapin) [inline src/atom/mod.rs]: This is only a win over using `str::to_owned` through `Deref<Target=str>` when the refcount happens to be 1, which seems seriously improbable. This this adds unsafe code, I’m mildly opposed to adding it without demonstrated perf wins.

Comment (nox) [inline src/atom/mod.rs]: It makes a round-trip free, but I guess we can remove it for now.

Comment (SimonSapin) [inline src/atom/mod.rs]: But do we ever do that?

**Pull request merged**